src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -771,10 +762,60 @@ void initHybridAPI() {
   CsvAccess(gpu_manager).init();
 }
 
-// Set HAPI device for non-0 ranks
-void setHybridAPIDevice() {
-  // set which device to use
-  hapiCheck(cudaSetDevice(getMyCudaDevice(CmiMyNode())));
+// Set up PE to GPU mapping, invoked from all PEs
+// TODO: Support custom mappings
+void initDeviceMapping(char** argv) {
+  int map_type = 0; // 0: block, 1: round-robin (default is block mapping)
```

Comment:
A nicer way might be to make this into a `class enum` (especially if there are more mapping in the future), but not sure if it is worth it here.